### ISSUE TITLE
 Fails to build under Ruby 2.4.0 (duplicate but with solution)

### ISSUE THREAD

**Issue by matkoniecz** [backup/backup#953]
use `gem install backup -v5.0.0.beta.1` - found thanks to https://github.com/backup/backup/issues/944#issuecomment-517833152

duplicate of #820

```
ERROR:  Error installing backup:
	ERROR: Failed to build gem native extension.

    current directory: /var/lib/gems/2.5.0/gems/json-1.8.2/ext/json/ext/generator
/usr/bin/ruby2.5 -I /usr/local/lib/site_ruby/2.5.0 -r ./siteconf20191012-6006-zifncm.rb extconf.rb
creating Makefile

current directory: /var/lib/gems/2.5.0/gems/json-1.8.2/ext/json/ext/generator
make "DESTDIR=" clean

current directory: /var/lib/gems/2.5.0/gems/json-1.8.2/ext/json/ext/generator
make "DESTDIR="
compiling generator.c
generator.c: In function ‘generate_json’:
generator.c:867:25: error: ‘rb_cFixnum’ undeclared (first use in this function)
     } else if (klass == rb_cFixnum) {
                         ^
generator.c:867:25: note: each undeclared identifier is reported only once for each function it appears in
generator.c:869:25: error: ‘rb_cBignum’ undeclared (first use in this function)
     } else if (klass == rb_cBignum) {
                         ^
Makefile:241: recipe for target 'generator.o' failed
make: *** [generator.o] Error 1

make failed, exit code 2
```
